solarforecastarbiter/metrics/deterministic.py

```diff
@@ -209,3 +213,113 @@ def centered_root_mean_square(y_true, y_pred):
     return np.sqrt(np.mean(
         ((y_pred - np.mean(y_pred)) - (y_true - np.mean(y_true))) ** 2
     ))
+
+
+def kolmogorov_smirnov_integral(y_true, y_pred, normed=False):
+    """Kolmogorov-Smirnov Test Integral (KSI).
+
+    Parameters
+    ----------
+    y_true : array-like
+        True values.
+    y_pred : array-like
+        Predicted values.
+    normed : bool, optional
+        If True, return the normalized KSI [%].
+
+    Returns
+    -------
+    ksi : float
+        The KSI between the true and predicted values.
+
+    Notes
+    -----
+        The calculation of the empirical CDF uses a right endpoint rule (the
+        default of the statsmodels ECDF function). For example, if the data is
+        [1.0, 2.0], then ECDF output is 0.5 for any input less than 1.0.
+
+    """
+
```

Comment:
does this function do the same thing as [`scipy.stats.ks_2samp`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ks_2samp.html)?

Reply:
Even if the answer is yes, perhaps it's good to keep this implementation for the consistency with the OVER implementation, which I don't see in scipy.

Reply:
This is something I'm trying to confirm with the references in the original metrics document. It seems like the KSI is different than KS test (both the definition and interpretation), but I'll verify.

Reply:
`scipy.stats.ks_2samp` isn't the same as the KSI metric. The statistic is a maximum difference rather than an integral of differences as in KSI.  Good thought though.

Reply:
Thanks for clarifying.